### PR TITLE
Azure features 

### PR DESCRIPTION
**Pull request by autoantwort**
At least one transport protocol must be selected.

### CODE REVIEW

**Comment by BillyONeal** [comment]
@LarryOsterman Does this look OK to you?

**Comment by LarryOsterman** [comment]
> @LarryOsterman Does this look OK to you?

Changes to the Azure Core vcpkg.json need to go through the azure-core-cpp repo rather than vcpkg. These changes will disappear as soon as we release azure core next.

Also this is a fairly significant breaking change to Azure Core, it shouldn't come through the vcpkg repo directly.

**Comment by antkmsft** [comment]
We have a PR that fixes dependencies when building `azure-core-cpp[core]`: https://github.com/Azure/azure-sdk-for-cpp/pull/4934

File: ports/azure-core-cpp/vcpkg.json [hunk 1]
```diff
@@ -20,23 +36,15 @@
...
       "host": true
-    },
-    {
-      "name": "wil",
-      "platform": "windows"
     }
   ],
   "default-features": [
-    "http"
+    "curl"
```
Comment (ahsonkhan) [inline ports/azure-core-cpp/vcpkg.json:42]: As it stands, we don't want `curl` to be the default-feature out of the box.

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Please file an issue on [azure-sdk-for-cpp](https://github.com/Azure/azure-sdk-for-cpp/issues) for the issue you are facing and let's see if we can improve the experience. The current `default-features` are by design.

**Comment by autoantwort** [comment]
Replaced by #34291

**Pull request closed without merging**